### PR TITLE
implement cancer summary with Victory

### PR DESCRIPTION
**Pull request by alisman**
image

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: end-to-end-tests/specs/screenshot.spec.js [hunk 1]
```diff
@@ -1,76 +1,73 @@
...
-};
-
 function runResultsTests(){
 
     it('render the oncoprint', function(){
         waitForOncoprint();
         browser.pause(2000);
         var res = browser.checkElement('#oncoprint');
-        ssAssert(res);
+        assertScree(res);
```
Comment (onursumer) [inline end-to-end-tests/specs/screenshot.spec.js]: shouldn't it be `assertScreenShotMatch`?

**Comment by ersinciftci** [comment]
Can we move it under `pages/resultsView`?

File: end-to-end-tests/specs/screenshot.spec.js (hunk 1 above)
Comment (alisman) [reply end-to-end-tests/specs/screenshot.spec.js]: hah. yes.

File: src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        const self = this;
+        return [{
+            target: "data",
+            eventHandlers: {
+                onMouseEnter: () => {
+                    return [
+                        {
+                            target: "data",
+                            mutation: (props:any) => {
+                                console.log(props.x, props.y);
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx]: still have to tweaks this positioning stuff.

File: src/pages/resultsView/cancerSummary/CancerSummaryContainer.tsx [hunk 3]
```diff
@@ -101,42 +134,15 @@ export default class CancerSummaryContainer extends React.Component<{ store: Res
     }
 
     public render() {
```
Comment (alisman) [inline src/pages/resultsView/cancerSummary/CancerSummaryContainer.tsx]: log of the rending stuff proved very painful to unit test so i focused on covering it in e2e's

**Comment by alisman** [comment]
@ersinciftci @adamabeshouse @onursumer test are not passing reliably on this but it's infrastructure related/build related, i'm fairly sure.

File: end-to-end-tests/lib/testUtils.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+var assert = require('assert');
+
+module.exports = {
+
+    assertScreenShotMatch(result, message){
+        assert(result[0].isWithinMisMatchTolerance);
```
Comment (adamabeshouse) [inline end-to-end-tests/lib/testUtils.js]: , message)

File: end-to-end-tests/specs/results.spec.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                browser.setValue("[data-test='alterationThresholdInput']",0)
+                browser.keys("Enter");
+            });
+
+            it('handles change to sample total threshold',function(){
+                browser.setValue("[data-test='sampleTotalThresholdInput']",312)
+                browser.keys("Enter");
+                var res = browser.checkElement('#pancancer_study_summary', { hide:['.qtip'] });
+                assertScreenShotMatch(res);
+                var res = browser.checkElement('#pancancer_study_summary', { hide:['.qtip'] });
```
Comment (adamabeshouse) [inline end-to-end-tests/specs/results.spec.js]: unused check?

File: end-to-end-tests/lib/testUtils.js (hunk 4 above)
Comment (alisman) [reply end-to-end-tests/lib/testUtils.js]: thanks

**Pull request merged**